tests/test_app.py

```diff
@@ -0,0 +1,7 @@
+import pytest
+from httpx import AsyncClient, codes
```

Comment:
Should we write everything with async tests using HTTPx?

https://fastapi.tiangolo.com/id/tutorial/testing/

General fastapi TestClient is enough

Reply:
We're `await`-ing the test client in all tests so they need to be async. Did you have something else in mind?

Reply:
If we are using fastapi test client, we needn't await right:

```
from fastapi.testclient import TestClient
.//import app as well

client = TestClient(app)

def test_read_main():
    response = client.get("/")
    assert response.status_code == 200
```

Reply:
But now I see you are looking to use the test_client setup in conftest itself. For that, yeah as you mentioned it needs to be awaited. LGTM 👍🏾. I was thinking of using `fastapi.testclient` as well to show users both way are actually possible.

No need to change though. It looks good to be merged 😄